toolkit/partner-chains-cli/src/config.rs

```diff
@@ -12,15 +12,34 @@ use std::fmt::{Display, Formatter, Write};
 use std::str::FromStr;
 use std::{marker::PhantomData, process::exit};
 
+/// Definition of a field in a config JSON with optional default value.
 pub struct ConfigFieldDefinition<'a, T> {
+	/// Config field name
 	pub name: &'a str,
+	/// Config file name
 	pub config_file: &'a str,
+	/// Path to config field in config file
 	pub path: &'a [&'a str],
+	/// Optional default value for config field
 	pub default: Option<&'a str>,
-	pub _marker: PhantomData<T>,
+	_marker: PhantomData<T>,
```

Comment:
Since you touch field visibilities, I think we don't need any `pub`. `pub(crate)` should be enough at most for anything but commands to be used by node.  But no need to change it in this PR.